File: app/src/main/java/com/kent/university/privelt/ui/dashboard/card/data_metrics/DataMetricsAdapter.java
package com.kent.university.privelt.ui.dashboard.card.data_metrics;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.kent.university.privelt.R;
import com.kent.university.privelt.model.CardItem;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class DataMetricsAdapter extends RecyclerView.Adapter<DataMetricsViewHolder> {

    private List<CardItem> dataMetrics;

    public DataMetricsAdapter() {
        this.dataMetrics = new ArrayList<>();
    }

    @NonNull
    @Override
    public DataMetricsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.cell_metrics, parent, false);
        return new DataMetricsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DataMetricsViewHolder holder, int position) {
        holder.bind(dataMetrics.get(position));
    }

    @Override
    public int getItemCount() {
        return dataMetrics.size();
    }

    public void setDataMetrics(List<CardItem> dataMetrics) {
        this.dataMetrics = dataMetrics;
    }
}

File: app/src/main/java/com/kent/university/privelt/ui/dashboard/card/data_metrics/DataMetricsAdapter.kt
/*
 *  This Source Code Form is subject to the terms of the Mozilla Public
 *  * License, v. 2.0. If a copy of the MPL was not distributed with this
 *  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.kent.university.privelt.ui.dashboard.card.data_metrics

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.kent.university.privelt.R
import com.kent.university.privelt.model.Card

class DataMetricsAdapter : RecyclerView.Adapter<DataMetricsViewHolder>() {
    private var card: Card? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DataMetricsViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.cell_metrics, parent, false)
        return DataMetricsViewHolder(view)
    }

    override fun onBindViewHolder(holder: DataMetricsViewHolder, position: Int) {
        holder.bind(card!!, card?.metrics?.get(position)!!)
    }

    override fun getItemCount(): Int {
        return card?.metrics?.size!!
    }

    fun setDataMetrics(card: Card) {
        this.card = card
    }
}